### DISCUSSION TITLE
Hi, how can I delay or pause all queries and resume them when needed?

### DISCUSSION THREAD

**Discussion by luckykenlin** [TanStack/query#7555] [Q&A]
Hi,

I use Laravel with react query and Axios as the fetching library. I set Axios withCredentials: true to avoid unauthorized issues.

Everything works great until I use Google login. 

After logging in with Google and being redirected back to the main page, I need to post the Google ID to the server for authentication. 

However, some stale queries are refetching in the background with unauthorized cookies,

as the login request hasn't yet been completed. This causes the stale queries to overwrite my cookie, leading to unauthorized issues.

Is there a flag I can use to pause all background queries and resume them when needed? 

This way, I can pause and resume all queries after the login process.

Any tips and help would be really appreciated.

**Comment by serdar-webdev** [comment]
Hi you can use cancelQueries and after refetchQueries, so in cancelQueries you canceling your queries basing on query keys, after you refetching queries by query key. I think it might work
